sideboard/websockets.py

```diff
@@ -452,11 +496,14 @@ def broadcast(cls, channels, trigger=None, originating_client=None):
         """
         triggered = set()
         for channel in sideboard.lib.listify(channels):
-            for websocket, clients in cls.subscriptions[channel].items():
-                for client, callbacks in clients.copy().items():
-                    if client != originating_client:
-                        for callback in callbacks:
-                            triggered.add((websocket, client, callback))
+            for websocket, clients in list(cls.subscriptions[channel].items()):
+                if websocket.is_closed:
+                    websocket.unsubscribe_all()
+                else:
+                    for client, callbacks in clients.copy().items():
+                        if client != originating_client:
+                            for callback in callbacks:
+                                triggered.add((websocket, client, callback))
```

Comment:
This entire change is a minor performance tweak.  When a websocket is closed, then there are two things which happen:
- the client sends a ``client-terminated`` message to the server
- the underlying socket connection is closed

In theory those two things should happen at about the same time.  In practice we've noticed that sometimes there will be a considerable period of time between those two events, especially when you're having Apache reverse-proxy your websocket connections.  Therefore, we now have Sideboard call ``unsubscribe_all()`` on any websocket connection which is marked as closed, since receiving ``client-terminated`` counts as closed even if the underlying socket connection is still open.  This prevents us from triggering update checks on subscriptions which can't be sent out anyway.

MAGFest doesn't currently care about this, though we've been talking about doing more websocket pub/sub in the future so this kind of thing will be relevant as soon as we do.